api/protobuf-spec/backend.proto

```diff
@@ -1,76 +1,56 @@
-// Follow the guidelines at https://cloud.google.com/endpoints/docs/grpc/transcoding
-// to keep the gRPC service definitions friendly to REST transcoding. An excerpt:
-//
-// "Transcoding involves mapping HTTP/JSON requests and their parameters to gRPC
-// methods and their parameters and return types (we'll look at exactly how you
-// do this in the following sections). Because of this, while it's possible to
-// map an HTTP/JSON request to any arbitrary API method, it's simplest and most
-// intuitive to do so if the gRPC API itself is structured in a
-// resource-oriented way, just like a traditional HTTP REST API. In other
-// words, the API service should be designed so that it uses a small number of
-// standard methods (corresponding to HTTP verbs like GET, PUT, and so on) that
-// operate on the service's resources (and collections of resources, which are
-// themselves a type of resource). 
-// These standard methods are List, Get, Create, Update, and Delete."
-//
 syntax = 'proto3';
+package api;
+option go_package = "github.com/GoogleCloudPlatform/open-match/internal/pb";
 
-service API {
+// The protobuf messages sent in the gRPC calls are defined 'messages.proto'.
+import 'api/protobuf-spec/messages.proto';
+
+service Backend {
   // Calls to ask the matchmaker to run a matchmaking function.
-  //
+
   // Run MMF once.  Return a matchobject that fits this profile.
-  rpc CreateMatch(Profile) returns (MatchObject) {} 
+  // INPUT: MatchObject message with these fields populated:
+  //  - id
+  //  - properties
+  //  - [optional] roster, any fields you fill are available to your MMF.
+  //  - [optional] pools, any fields you fill are available to your MMF.
+  // OUTPUT: MatchObject message with these fields populated:
+  //  - id
+  //  - properties
+  //  - error. Empty if no error was encountered
+  //  - rosters, if you choose to fill them in your MMF. (Recommended)
+  //  - pools, if you used the MMLogicAPI in your MMF. (Recommended, and provides stats)
+  rpc CreateMatch(messages.MatchObject) returns (messages.MatchObject) {} 
   // Continually run MMF and stream matchobjects that fit this profile until
-  // client closes the connection.
-  rpc ListMatches(Profile) returns (stream MatchObject) {}
+  // client closes the connection.  Same inputs/outputs as CreateMatch.
+  rpc ListMatches(messages.MatchObject) returns (stream messages.MatchObject) {}
```

Comment:
Otherwise it would cause some eyebrow raising when someone gets 200 match functions all hitting back at once even though it's expected :)

Reply:
If anything, that might be another reason to remove the evaluator sooner rather than later.  I'm not sure it really fits the paradigm as well once we've chosen not to allow that additional level of 'fan-out' within a profile (e.g. specifying multiple MMF images in a single profile).

Reply:
At some point you've to fan-in though? Unless you can guarantee that all match functions create deterministically different sets? I guess you could not care that players are in multiple matches, and that a dev has some kind of build a match, then continuously munge it type of workflow? Some kind of supported fan-in seems useful. Maybe we can break that out from the evaluator concept.